tests/unit/lms/views/lti/deep_linking_test.py

```diff
@@ -69,11 +69,38 @@ def test_it_for_v13(
                         "url": _get_content_url_mock.return_value,
                     }
                 ],
-                "https://purl.imsglobal.org/spec/lti-dl/claim/data": sentinel.deep_linking_settings,
+                "https://purl.imsglobal.org/spec/lti-dl/claim/data": sentinel.deep_linking_settings_data,
             }
         )
         assert fields["JWT"] == ltia_http_service.sign.return_value
 
+    def test_it_for_v13_missing_deep_linking_settings_data(
+        self,
+        ltia_http_service,
+        application_instance,
+        views,
+        _get_content_url_mock,
+        pyramid_request,
+    ):
+        del pyramid_request.parsed_params["deep_linking_settings"]["data"]
+
+        views.file_picker_to_form_fields_v13()
+
+        # https://purl.imsglobal.org/spec/lti-dl/claim/data is now not present in the message
+        ltia_http_service.sign.assert_called_once_with(
+            {
+                "https://purl.imsglobal.org/spec/lti/claim/deployment_id": application_instance.deployment_id,
+                "https://purl.imsglobal.org/spec/lti/claim/message_type": "LtiDeepLinkingResponse",
+                "https://purl.imsglobal.org/spec/lti/claim/version": "1.3.0",
+                "https://purl.imsglobal.org/spec/lti-dl/claim/content_items": [
+                    {
+                        "type": "ltiResourceLink",
+                        "url": _get_content_url_mock.return_value,
+                    }
+                ],
+            }
+        )
```

Comment:
```suggestion
    def test_it_for_v13_missing_deep_linking_settings_data(
        self, ltia_http_service, views, pyramid_request
    ):
        del pyramid_request.parsed_params["deep_linking_settings"]["data"]

        views.file_picker_to_form_fields_v13()

        ltia_http_service.sign.assert_called_once_with(message := Any.dict())
        assert (
            "https://purl.imsglobal.org/spec/lti-dl/claim/data "
            not in message.last_matched()
        )
```

Fancy new matcher stuff to the rescue.